inc/commondbrelation.class.php

```diff
@@ -1424,42 +1511,56 @@ static function processMassiveActionsForOneItemtype(MassiveAction $ma, CommonDBT
             foreach ($ids as $key) {
                // First, get the query to find all occurences of the link item<=>key
                if (!$peer) {
-                  $query = static::getSQLRequestToSearchForItem($item->getType(), $key);
+                  $criteria = static::getSQLCriteriaToSearchForItem($item->getType(), $key);
                } else {
                   if (!$item->getFromDB($key)) {
                      $ma->itemDone($item->getType(), $key, MassiveAction::ACTION_KO);
                      $ma->addMessage($item->getErrorMessage(ERROR_NOT_FOUND));
                      continue;
                   }
+
                   $query = 'SELECT `'.static::getIndexName().'`
                             FROM `'.static::getTable().'`';
-                  $WHERE = [];
+                  $WHERE = [
+                     static::$items_id_1  => $item_1->getID(),
+                     static::$items_id_2  => $item_2->getID()
+                  ];
                   if (preg_match('/^itemtype/', static::$itemtype_1)) {
-                     $WHERE[] = " `".static::$itemtype_1."` = '".$item_1->getType()."'";
+                     $WHERE[static::$itemtype_1] = $item_1->getType();
                   }
-                  $WHERE[] = " `".static::$items_id_1."` = '".$item_1->getID()."'";
                   if (preg_match('/^itemtype/', static::$itemtype_2)) {
-                     $WHERE[] = " `".static::$itemtype_2."` = '".$item_2->getType()."'";
+                     $WHERE[static::$itemtype_2] = $item_2->getType();
                   }
-                  $WHERE[] = " `".static::$items_id_2."` = '".$item_2->getID()."'";
-                  $query .= 'WHERE ('.implode(' AND ', $WHERE).')';
 
                   if (($specificities['check_both_items_if_same_type'])
                       && ($item_1->getType() == $item_2->getType())) {
-                     $WHERE = [];
+                     $ORWHERE = [
+                        static::$items_id_1 = $item_2->getID(),
+                        static::$items_id_2 = $item_2->getID()
+                     ];
                      if (preg_match('/^itemtype/', static::$itemtype_1)) {
-                        $WHERE[] = " `".static::$itemtype_1."` = '".$item_2->getType()."'";
+                        $ORWHERE[static::$itemtype_1] = $item_2->getType();
                      }
-                     $WHERE[] = " `".static::$items_id_1."` = '".$item_2->getID()."'";
                      if (preg_match('/^itemtype/', static::$itemtype_2)) {
-                        $WHERE[] = " `".static::$itemtype_2."` = '".$item_2->getType()."'";
+                        $ORWHERE[static::$itemtype_2] = $item_2->getType();
                      }
-                     $WHERE[] = " `".static::$items_id_2."` = '".$item_2->getID()."'";
-                     $query .= ' OR ('.implode(' AND ', $WHERE).')';
+                     $WHERE = [
+                        'OR' => [
+                           $WHERE,
```

Comment:
`AND` operator have to added here to have `OR (a AND b AND c)` here.
```suggestion
                           ['AND' => $WHERE],
```